app/services/BlackFriday.scala

```diff
@@ -0,0 +1,55 @@
+package services
+
+import configuration.Config
+import model.DigitalEdition
+import model.promoCodes.{GuardianWeekly, _}
+import org.joda.time.DateTime
+
+//TODO: Remove after 3rd December 2017
```

Comment:
So all that's needed to tidy things up after it self-deactivates is to delete this object and just remove any usage of it throughout the templates? i.e. there's no need to replace it with anything?

If so, if this is made an implicit class of something, and is forcibly the only overriding/implicit class, can we make it so that just deleting the class will work in future, and no need to modify the templates?

Reply:
It will require a bit more refactoring than that because this object now provides the promocodes that are passed through the templates. I could have moved that responsibility into a separate class and had this just deal with whether the promotion was running or not, but even then that other object would have had to have had some awareness of the promotion so it seemed easier just to put it in here for now.